UnitTests/GitCommandsTests/GitRevisionTests.cs

```diff
@@ -19,5 +20,34 @@ public void Should_validate_full_sha1_correctly()
             Assert.False(GitRevision.IsFullSha1Hash("0000000000000000000000000000000000000000 "));
             Assert.False(GitRevision.IsFullSha1Hash(" 0000000000000000000000000000000000000000"));
         }
+
+        [TestCase(null)]
+        [TestCase("toto\n")]
+        [TestCase("toto\ntata\ntiti\n")]
+        public void Should_have_same_content_When_no_ellipsis(string bodyContent)
+        {
+            var gitRevision = new GitRevision(ObjectId.Random());
+            gitRevision.Body = bodyContent;
+            gitRevision.HasMultiLineMessage = true;
+            Assert.AreEqual(bodyContent, gitRevision.BodySummary);
+        }
+
+        [Test]
+        public void Should_do_ellipsis_When_too_many_lines()
+        {
+            var gitRevision = new GitRevision(ObjectId.Random());
+            gitRevision.Body = "line1\nline2\nline3\nline4\nline5\nline6\nline7\nline8\nline9\nline10\nline11\nline12\nline13\nline14\nline15\nline16\nline17\nline18\nline19\nline20\nline21\nline22\nline23\nline24\nline25\nline26\nline27\nline28\nline29\nline30\nline31";
+            gitRevision.HasMultiLineMessage = true;
+            Assert.AreEqual("line1\nline2\nline3\nline4\nline5\nline6\nline7\nline8\nline9\nline10\nline11\nline12\nline13\nline14\nline15\nline16\nline17\nline18\nline19\nline20\nline21\nline22\nline23\nline24\nline25\nline26\nline27\nline28\nline29\nline30\n[...]", gitRevision.BodySummary);
```

Comment:
Let's convert these test (that deal with blobs of text) into using ApprovalTests.
It would make it easier to generate blobs of texts and visually assert results.

Have a look at https://github.com/gitextensions/gitextensions/blob/94621eaac6ed91aae3cba072361710c3326823ed/UnitTests/GitUITests/Editor/FileViewerTextTests.cs#L47-L66